### PR TITLE
new resource: `azurerm_spring_cloud_new_relic_application_performance_monitoring`

### PR DESCRIPTION
**Pull request by ms-henglu**
```
=== RUN   TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_basic
=== PAUSE TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_basic
=== CONT  TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_basic
=== RUN   TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_requiresImport
=== PAUSE TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_requiresImport
=== CONT  TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_requiresImport
=== RUN   TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_complete
=== PAUSE TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_complete
=== CONT  TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_complete
=== RUN   TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_update
=== PAUSE TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_update
=== CONT  TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_update
--- PASS: TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_requiresImport (542.21s)
--- PASS: TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_basic (555.97s)
--- PASS: TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_complete (637.54s)
--- PASS: TestAccSpringCloudNewRelicApplicationPerformanceMonitoring_update (693.14s)
PASS

Process finished with the exit code 0
```

### CODE REVIEW

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource.go [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model SpringCloudNewRelicApplicationPerformanceMonitoringModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			client := metadata.Client.AppPlatform.AppPlatformClient
+			springId, err := commonids.ParseSpringCloudServiceID(model.SpringCloudServiceId)
+			if err != nil {
+				return fmt.Errorf("parsing spring service ID: %+v", err)
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource.go:138]:
```suggestion
				return err
```

File: website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  app_name                        = "example-app-name"
+  license_key                     = "example-license-key"
+  agent_enabled                   = true
+  app_server_port                 = 8080
+  audit_mode_enabled              = true
+  auto_app_naming_enabled         = true
+  auto_transaction_naming_enabled = true
+  custom_tracing_enabled          = true
+  labels                          = "tagName1:tagValue1;tagName2:tagValue2;tagName3:tagValue3"
+  globally_enabled                = true
```
Comment (magodo) [inline website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown:46]: These boolean properties all have default values, so can we omit them in the example to make it simple?

File: website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+* `audit_mode_enabled` - (Optional) Specifies whether enable plain text logging of all data sent to New Relic to the agent logfile. Defaults to `false`.
+
+* `auto_app_naming_enabled` - (Optional) Specifies whether enable the reporting of data separately for each web app. Defaults to `false`.
+
+* `auto_transaction_naming_enabled` - (Optional) Specifies whether enable the component-based transaction naming. Defaults to `true`.
+
+* `custom_tracing_enabled` - (Optional) Specifies whether enable all instrumentation using an `@Trace` annotation. Disabling this causes `@Trace` annotations to be ignored. Defaults to `true`.
+
+* `labels` - (Optional) Specifies the labels to be added to the New Relic application. The format is `tagName1:tagValue1;tagName2:tagValue2;tagName3:tagValue3`.
```
Comment (magodo) [inline website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown:76]: Can we make this to be a list instead?

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+func (r SpringCloudNewRelicApplicationPerformanceMonitoringResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_spring_cloud_new_relic_application_performance_monitoring" "test" {
+  name                            = "acctest-apm-%[2]d"
+  spring_cloud_service_id         = azurerm_spring_cloud_service.test.id
+  app_name                        = "test-app-name"
+  license_key                     = "test-license-key"
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:161]: Can we change these values to be different than `basic`?

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource.go [hunk 5]
```diff
@@ -0,0 +1,396 @@
...
+		"labels": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"globally_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			Default:  false,
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource.go:117]:
```suggestion
```

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  agent_enabled                   = true
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:162]: Same here

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+  app_server_port                 = 8080
+  audit_mode_enabled              = true
+  auto_app_naming_enabled         = true
+  auto_transaction_naming_enabled = true
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:166]: Same here

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+  custom_tracing_enabled          = true
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:167]: Same here

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

**Comment by ms-henglu** [comment]
Thanks @magodo ! I've updated it as suggested.

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go [hunk 9]
```diff
@@ -157,16 +157,19 @@ func (r SpringCloudNewRelicApplicationPerformanceMonitoringResource) complete(da
...
   auto_app_naming_enabled         = true
-  auto_transaction_naming_enabled = true
-  custom_tracing_enabled          = true
-  labels                          = "tagName1:tagValue1;tagName2:tagValue2;tagName3:tagValue3"
-  globally_enabled                = true
+  auto_transaction_naming_enabled = false
+  custom_tracing_enabled          = false
+  labels = {
+    "tagName1" = "tagValue1"
+    "tagName2" = "tagValue2"
```
Comment (magodo) [inline internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go:170]: The keys can be unquoted?

File: website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown [hunk 10]
```diff
@@ -32,18 +32,16 @@ resource "azurerm_spring_cloud_service" "example" {
...
-  labels                          = "tagName1:tagValue1;tagName2:tagValue2;tagName3:tagValue3"
-  globally_enabled                = true
+  name                    = "example"
+  spring_cloud_service_id = azurerm_spring_cloud_service.example.id
+  app_name                = "example-app-name"
+  license_key             = "example-license-key"
+  app_server_port         = 8080
+  labels = {
+    "tagName1" = "tagValue1"
+    "tagName2" = "tagValue2"
```
Comment (magodo) [inline website/docs/r/spring_cloud_new_relic_application_performance_monitoring.html.markdown:42]: Same here

**Review by magodo — CHANGES_REQUESTED** [review]
Some minor comments, which LGTM once fixed.

File: internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go (hunk 9 above)
Comment (ms-henglu) [reply internal/services/springcloud/spring_cloud_new_relic_application_performance_monitoring_resource_test.go]: Thanks! I've fixed it.

**Comment by magodo** [comment]
LGTM!

**Review by katbyte — APPROVED** [review]
LGTM me too 🚡

**Pull request merged**